fuse_graphs/src/hash_graph.cpp

```diff
@@ -409,6 +409,22 @@ ceres::Solver::Summary HashGraph::optimize(const ceres::Solver::Options& options
   return summary;
 }
 
+void HashGraph::print(std::ostream& stream) const
+{
+  stream << "HashGraph\n"
+         << "  constraints:\n";
+  for (const auto& constraint : constraints_)
+  {
+    stream << "   - " << *constraint.second << "\n";
```

Comment:
Is there any way to pass some extra indentation to `*constraint.second`? (same for `*variable.second` below)

The main motivation to have proper indentation is that it looks like we could get YAML compliant outputs that could be use to visualize the graph with graphviz if we create a YAML to dot translator. I wonder if that was your plan?

Reply:
>Is there any way to pass some extra indentation to *constraint.second? (same for *variable.second below)

Not currently. We could modify the `print()` function to accept an indentation string. The problem with that being that there is no way to set that parameter when using the stream operator. We could drop the stream operator in favor of a `toString()` method. Then you could:
```
std::cout << constraint.toString("    ") << std::endl;
```

> we could get YAML compliant outputs

That was not something I had thought of. If the output looks like YAML, it is pure coincidence. Returning valid YAML is interesting. The one downside I see is formatting the matrices. I was planning on updating the covariance output to use Eigen's [IOFormatter](https://eigen.tuxfamily.org/dox/structEigen_1_1IOFormat.html), to get pretty looking matrices. But it may be possible to create pretty valid YAML matrices and we both win.

Reply:
Regarding `toString`, although it sounds like the only option, I'm afraid that returning an `std::string` would be slower than using streams, which IIRC would manage things efficiently in terms of memory and IO.

Re. Eigen matrix, I think there should be an IOFormatter that should be YAML compliant. I could give that idea a try on another PR.